# Generated by Django 3.2.15 on 2023-03-20 09:48

import datetime
import django.core.validators
from django.db import migrations, models


class Migration(migrations.Migration):

    dependencies = [
        ('dairy', '0002_alter_cow_options_alter_lactation_options_and_more'),
    ]

    operations = [
        migrations.AlterModelOptions(
            name='insemination',
            options={'ordering': ['-date_of_insemination'], 'verbose_name': 'AI Record ðŸ“†', 'verbose_name_plural': 'AI Records ðŸ“†'},
        ),
        migrations.RemoveField(
            model_name='pregnancy',
            name='is_pregnant',
        ),
        migrations.AlterField(
            model_name='cow',
            name='date_of_birth',
            field=models.DateField(error_messages={'max_value': 'The date of birth cannot be in the future!.'}, validators=[django.core.validators.MaxValueValidator(datetime.date(2023, 3, 20))]),
        ),
        migrations.AlterField(
            model_name='cow',
            name='date_of_death',
            field=models.DateField(blank=True, error_messages={'max_value': 'The date of death cannot be in the future!.'}, null=True, validators=[django.core.validators.MaxValueValidator(datetime.date(2023, 3, 20))]),
        ),
        migrations.AlterField(
            model_name='insemination',
            name='date_of_insemination',
            field=models.DateField(error_messages={'max_value': 'Invalid date entry, Dates must not be in future'}, validators=[django.core.validators.MaxValueValidator(datetime.date(2023, 3, 20))]),
        ),
        migrations.AlterField(
            model_name='pregnancy',
            name='pregnancy_failed_date',
            field=models.DateField(blank=True, error_messages={'max_value': 'Future records not allowed!.'}, null=True, validators=[django.core.validators.MaxValueValidator(datetime.date(2023, 3, 20))]),
        ),
        migrations.AlterField(
            model_name='pregnancy',
            name='pregnancy_outcome',
            field=models.CharField(blank=True, choices=[('Live', 'Live'), ('Stillborn', 'Stillborn'), ('Miscarriage', 'Miscarriage')], max_length=11, null=True),
        ),
        migrations.AlterField(
            model_name='pregnancy',
            name='pregnancy_status',
            field=models.CharField(choices=[('Confirmed', 'Confirmed'), ('Unconfirmed', 'Unconfirmed'), ('Failed', 'Failed')], default='Unconfirmed', max_length=11),
        ),
        migrations.AlterField(
            model_name='semen',
            name='date_of_expiry',
            field=models.DateField(error_messages={'min_value': 'Invalid date entry, Date of expiry must be in future'}, validators=[django.core.validators.MinValueValidator(datetime.date(2023, 3, 20))]),
        ),
        migrations.AlterField(
            model_name='semen',
            name='date_of_production',
            field=models.DateField(error_messages={'max_value': 'Invalid date entry, Dates of production must not be in future'}, validators=[django.core.validators.MaxValueValidator(datetime.date(2023, 3, 20))]),
        ),
        migrations.AlterField(
            model_name='symptoms',
            name='date_observed',
            field=models.DateField(error_messages={'max_value': 'The date of observation cannot be in the future!.'}, validators=[django.core.validators.MaxValueValidator(datetime.date(2023, 3, 20))]),
        ),
        migrations.AlterUniqueTogether(
            name='insemination',
            unique_together={('cow', 'date_of_insemination')},
        ),
    ]
